test/library/packages/Socket/tcpio.chpl

```diff
@@ -0,0 +1,96 @@
+use Socket;
+use UnitTest;
+use IO;
+
+proc recv_string(test: borrowed Test) throws {
+  var port:uint(16) = 8811;
+  var address = ipAddr.ipv4(IPv4Localhost, port);
+  var server = listen(address);
+  sync {
+    begin {
+      var conn = server.accept();
+      var writer = conn.writer();
+      writer.write("hello world\n");
+    }
+
+    var conn = connect(address);
+    var reader = conn.reader();
+    var x:bytes;
+    reader.readline(x);
+    test.assertEqual(x, b"hello world\n");
+  }
+}
+
+proc recv_number(test: borrowed Test) throws {
+  var port:uint(16) = 8810;
+  var address = ipAddr.ipv4(IPv4Localhost, port);
+  var server = listen(address);
+  sync {
+    begin {
+      var conn = server.accept();
+      var writer = conn.writer();
+      writer.write(42, " ");
+    }
+
+    var conn = connect(address);
+    var reader = conn.reader();
+    var y:int = reader.read(int);
+    test.assertEqual(y, 42);
+  }
+}
+
+proc send_string(test: borrowed Test) throws {
+  var port:uint(16) = 6000;
+  var address = ipAddr.ipv6(IPv6Localhost, port);
+  var server = listen(address);
+  sync {
+    begin {
+      var conn = server.accept();
+      var reader = conn.reader();
+      var x:bytes;
+      reader.readline(x);
+      test.assertEqual(x, b"hello world\n");
+    }
+
+    var conn = connect(address);
+    var writer = conn.writer();
+    writer.write("hello world\n");
+  }
+}
+
+proc send_number(test: borrowed Test) throws {
+  var port:uint(16) = 6000;
+  var address = ipAddr.ipv4(IPv4Localhost, port);
+  var server = listen(address);
+  sync {
+    begin {
+      var conn = server.accept();
+      var reader = conn.reader();
+      var y:int = reader.read(int);
+      test.assertEqual(y, 42);
+    }
+
+    var conn = connect(address);
+    var writer = conn.writer();
+    writer.write(42, " ");
+  }
+}
```

Comment:
@mppf I made a local dev env with Ubuntu and tried to work out the issue that I observed in VM. 

It turns this is a pretty flaky error and is not caused by `sent_http` it's by `send_number`. This was something I observed earlier too but I thought was resolved at that time so if you run the test 4/5 times it might run perfectly through all or will be hangup randomly at this test.

The thing that is intriguing is if I remove the `recv_number` test everything works fine without hanging even if you run it 1k times.